bionic/datatypes.py

```diff
@@ -302,6 +305,10 @@ class CodeVersion:
 
     major: str = attr.ib(converter=str_from_version_value)
     minor: str = attr.ib(converter=str_from_version_value)
+    ignore_bytecode: bool = attr.ib(converter=attr.converters.default_if_none(False))
```

Comment:
They aren't part of the "version" per se, but they are ingested into Bionic from `@version` decorator that directly translates to `CodeVersion`.

Also, even though `suppress_bytecode_warnings` doesn't define the version, `ignore_bytecode` does. For automatic version, the provenance hash of a function with version `1.0` will be different when `ignore_bytecode` is `True` vs `False`.

Reply:
Yes, they are ingested through `@version`, but our API doesn't need to be 1:1 with our internal data model. And I would argue that neither `suppress_bytecode_warnings` nor `ignore_bytecode` are really part of the version; they're per-entity configuration options that affect how we interpret the version. For example, if someone asked "what is the version of this entity", I would expect the answer to include the major and minor numbers (maybe as a human-readable string), but not the `ignore_bytecode` settings. Similarly, it doesn't line up with what's described in the docstring of `CodeVersion`.

Of course, that's not to say that we can't stick these on `CodeVersion` anyway to make things simpler. But since this class gets serialized as part of our cache schema, it will be annoying to change later, so I just want to make sure we're thoughtful about why we're doing this.

Reply:
After thinking about this some more, I've partly come around to your point of view here: maybe it does make sense for these fields to be here. Here's my current thinking:

1. Having things that look like "instructions" on a version object seems weird to me. I feel like a version should be more like a piece of data that gets interpreted rather than a set of instructions. However, if we think of `ignore_bytecode` as more like `includes_bytecode` (changing from an instruction to a description) then it makes more sense to me. This is similar to the difference between `@persist` and `@serializable`.

2. I don't have a similar rationale for `suppress_bytecode_warnings` but you can't win them all. :)

Reply:
I guess another possibility would be to have a `CodeVersioningPolicy` object that includes a `CodeVersion` along with all the additional settings.

Reply:
I think I'll do both, change `ignore_bytecode` to `includes_bytecode` and create a `CodeVersioningPolicy` object. I still think `includes_bytecode` should be a part of code version, but it would be good to have a policy object to contain other fields.